src/screens/Messaging/Chatrooms/ChatroomsScreen.js

```diff
@@ -19,61 +19,47 @@ export default function ChatroomsScreen({ navigation }) {
   const uid = navigation.getParam('uid', null);
   const [chats, setChats] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
-  const chatRoomRef = Firestore().collection('chatrooms');
 
   useEffect(() => {
     // eslint-disable-next-line no-console
-    console.log('Mounted. Retrieving chatrooms...');
-    setIsLoading(true);
-    setChats([]); // Reset for when react native auto remounts in emulator
-    // chatRoomRef.add({
-    //   users: [uid, '123'],
-    //   names: { uid: 'tester', 123: 'test123' },
-    //   createdAt: Firestore.FieldValue.serverTimestamp(),
-    //   updatedAt: Firestore.FieldValue.serverTimestamp(),
-    // }).then((newchatRef) => {
-    //   newchatRef.collection('messages').add({
-    //     senderId: uid,
-    //     text: 'test message',
-    //     sentAt: Firestore.FieldValue.serverTimestamp(),
-    //   });
-    //   newchatRef.collection('messages').add({
-    //     senderId: '123',
-    //     text: 'test received message',
-    //     sentAt: Firestore.FieldValue.serverTimestamp(),
-    //   });
-    // });
-    try {
-      // build chats - list of tuples of chatrooms with (recipients, updatedAt, messages)
-      // |- recipients is a map of userid: displayname
-      chatRoomRef.where('users', 'array-contains', uid).orderBy('updatedAt', 'desc')
-        .get()
-        .then((querySnapshot) => {
-          querySnapshot.forEach((doc) => {
-            const users = doc.data().names;
-            delete users[uid]; // exclude own name
-            // eslint-disable-next-line no-shadow
-            setChats((chats) => [...chats, {
-              recipients: users,
-              updatedAt: doc.data().updatedAt,
-              messages: doc.ref.collection('messages'),
-            }]);
-          });
-        })
-        .then(() => {
+    console.log('Mounted. Subscribing to chatrooms...');
+    // build chats - list of tuples of chatrooms with (recipients, updatedAt, messages)
+    // |- recipients is a map of userid: displayname
+    const unsubscribe = firestore()
+      .collection('chatrooms')
+      .where('users', 'array-contains', uid)
+      .orderBy('updatedAt', 'desc')
```

Comment:
might want to set an arbitrary limit here so we dont end up pulling too many chatrooms if the user has a lot, and scroll to view more